frontend_app_file/src/component/Kanban.jsx

```diff
@@ -0,0 +1,386 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import { v4 as uuidv4 } from 'uuid'
+import Board from '@lourenci/react-kanban'
+import '@lourenci/react-kanban/dist/styles.css'
+import {
+  IconButton,
+  PromptMessage,
+  handleFetchResult,
+  putFileContent,
+  removeLocalStorageItem,
+  getLocalStorageItem,
+  setLocalStorageItem
+} from 'tracim_frontend_lib'
+
+require('./Kanban.styl')
+
+function moveElem (array, fromPosition, toPosition, elem) {
+  const a = array.slice()
+  a.splice(fromPosition, 1)
+  a.splice(toPosition, 0, elem)
+  return a
+}
+
+export default translate()(class Kanban extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      loading: false,
+      mustSave: false,
+      board: {
+        columns: [],
+        ...JSON.parse(props.content.raw_content || '{}')
+      },
+      isDrafAvailable: !!getLocalStorageItem(
+        'rawContent',
+        props.content,
+        'kanban'
+      )
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    if (this.props.content.current_revision_id !== prevProps.content.current_revision_id) {
+      // TODO handle the "Refresh?" button
+      this.updateBoard({
+        ...this.state.board,
+        ...JSON.parse(this.props.content.raw_content || '{}')
+      }, true)
+    }
+  }
+
+  handleRestoreDraft = () => {
+    this.updateBoard(
```

Comment:
update board does a setState() and there is another one bellow
It is a good practice in react to try and factorize the setState to avoid multiple rerenders and side effects
Here, your updateBoard() could return the new board or you could pass any other state update to updateBoard
Those are examples, I haven't checked if they are good ideas

Reply:
React will batch state changes: https://stackoverflow.com/questions/33613728/what-happens-when-using-this-setstate-multiple-times-in-react-component

I'd rather keep a separation of concerns: each function does what it should do. But we can discuss.

Reply:
the issue with setState batching is that you don't handle it. React does it for perf optimization but you can't know which setStates are going to be batched.
Therefore, you can't be sure that the ones you expect to be batched always will be. And if they don't, there will 2 rerenders that might create unexpected behavior.

I agree with the separation of concerns though, my advise would be to have a function that act as a controller that calls separate functions that each returns their own new state part. And the "controller" do a final setState with the returns of every separated functions